src/test/java/io/confluent/clientmetrics/SchemaRegistryContainer.java

```diff
@@ -0,0 +1,29 @@
+package io.confluent.clientmetrics;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.Network;
+
+public class SchemaRegistryContainer extends GenericContainer<SchemaRegistryContainer> {
```

Comment:
Are you aware of the mock SR we ship?

Reply:
Yes, but it is a pain to configure serdes for the mock SR client.... =(

```java
        final SchemaRegistryClient schemaRegistry = new MockSchemaRegistryClient();

        StringSerializer keySerializer = new StringSerializer(schemaRegistry); // TODO doesn't take the client as a parameter
        SpecificAvroSerializer<AVROMetricRecords> valSerializer = new SpecificAvroSerializer<>(schemaRegistry); // TODO doesn't take the client as a parameter

        final KafkaProducer<String, AVROMetricRecords> producer = new KafkaProducer<>(
                Map.of(
                        ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName(),
                        ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class.getName(),
                        KafkaAvroSerializerConfig.SCHEMA_REGISTRY_URL_CONFIG, "http://mock:8081",
                        ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers(),
                        ProducerConfig.CLIENT_ID_CONFIG, UUID.randomUUID().toString()
                ), keySerializer, valSerializer);
```

but no such problems with testcontainers.

Reply:
yeeeah, but you can reuse that code, and docker is ~slow